### PR TITLE
New component: Badge

### PR DESCRIPTION
**Pull request by mqtik**

### CODE REVIEW

File: src/components/badge/badge.test.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { matchers } from '@emotion/jest';
+import Badge from './badge';
+import BadgeProps from './badge.types';
+import { ThemeProvider } from '@emotion/react';
+import userEvent from '@testing-library/user-event';
```
Comment (BrandonDenham) [inline src/components/badge/badge.test.tsx:7]: seems to be alot of unused imports

File: src/components/badge/badge.test.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import theme from '../../shared/themes/theme';
+
+expect.extend(matchers);
+
+describe('Badge', () => {
```
Comment (BrandonDenham) [inline src/components/badge/badge.test.tsx:13]: so should actually write some of the tests

File: src/components/badge/badge.styles.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { css, Theme } from '@emotion/react';
+
+export const mainContainer = (backgroundColor: string, borderRadius: number) => css`
```
Comment (BrandonDenham) [inline src/components/badge/badge.styles.tsx:3]: lets name this badgeContainer

File: src/components/badge/badge.styles.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    position: relative;
+    display: inline-block;
+    minWidth: 20px;
+    minHeight: 20px;
+    padding: 10px;
+    ${backgroundColor && `background-color: ${backgroundColor};`}
+    ${borderRadius && `border-radius: ${borderRadius}px;`}
+`;
+
+export const detailSpan = (color: string,fontSize: number) => css`
```
Comment (BrandonDenham) [inline src/components/badge/badge.styles.tsx:13]: lets call it badgeText

File: src/components/badge/badge.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
+/** @jsx jsx */
```
Comment (BrandonDenham) [inline src/components/badge/badge.tsx:1]: fileName  should match default export (so be capitalized)

File: src/components/loader/_private/componentLoader.tsx [hunk 6]
```diff
@@ -79,7 +79,7 @@ const createMarkup = ({
     };
 };
 
-const Loader: React.FC<LoaderProps> = props => {
+const Loader: React.FC<LoaderProps> = (props: any) => {
```
Comment (BrandonDenham) [inline src/components/loader/_private/componentLoader.tsx:82]: props should be LoaderPros, not any

File: src/components/badge/badge.types.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { FieldProps } from '../field/field.types';
+
+export default interface BadgeProps {
+    label?: string;
+    backgroundColor?: string;
+    color?: string;
```
Comment (BrandonDenham) [inline src/components/badge/badge.types.ts:6]: the props should be the props from the approved design:  https://trovatrip.slite.com/api/s/note/Wsdf6nLehQSJUhhoXb2TaK/Badge  

label should be text and it should be required.  

borderRadius and fontSize should NOT be exposed, please remove

onClick and color are OK (so add them to the design doc), but color should be an enum of our colors, onClick should be a function, not any (NEVER any) 

if it's not listed in the api in the design doc, it shouldn't be here.   These things shouldn't come up in code review but on design doc comments where you ping the members of the team.

File: src/components/badge/badge.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+const Badge: React.FC<BadgeProps> = ({
+    onClick,
+    backgroundColor = '#00b3ca',
+    color = '#ffffff',
+    borderRadius = 4,
+    fontSize = 20,
+    label,
+}) => {
+    const theme = useTheme();
```
Comment (BrandonDenham) [inline src/components/badge/badge.tsx:24]: unused?

File: src/components/badge/badge.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    const handleBadge = useCallback(
```
Comment (BrandonDenham) [inline src/components/badge/badge.tsx:25]: handleBadgeClick

File: src/components/badge/badge.tsx [hunk 10]
```diff
@@ -0,0 +1,46 @@
+/** @jsx jsx */
+import { jsx, useTheme } from '@emotion/react';
+import React, { SyntheticEvent, useCallback } from 'react';
+
+import BadgeProps from './badge.types';
+import { mainContainer, detailSpan } from './badge.styles';
+import ComponentFooter from '../__private/componentFooter';
```
Comment (BrandonDenham) [inline src/components/badge/badge.tsx:7]: unused?

**Review by BrandonDenham — CHANGES_REQUESTED** [review]

File: src/components/badge/badge.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+        color="#ffffff"
+        onClick={() => {}}
+    />
+);
+
+export const BadgeNumber = () => (
+    <Badge
+        label="12"
+        backgroundColor="#00b3ca"
+        borderRadius={50}
```
Comment (BrandonDenham) [inline src/components/badge/badge.stories.tsx:24]: can delete the non-used props

File: src/components/badge/badge.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+import React, { SyntheticEvent, useCallback } from 'react';
+
+import BadgeProps from './badge.types';
+import { badgeContainer, badgeText } from './badge.styles';
+
+/**
+ * Renders a <Badge /> component
+ * @param  props
+ * @param  props.label - Label
+ * @param  props.backgroundSize - backgroundSize
```
Comment (BrandonDenham) [inline src/components/badge/badge.tsx:12]: these should be accurate

File: src/components/badge/badge.types.ts [hunk 13]
```diff
@@ -0,0 +1,10 @@
+import { Colors } from '../../shared/constants/colors';
+import { SyntheticEvent } from 'react';
+export default interface BadgeProps {
+    label?: string;
+    backgroundColor?: string;
+    color?: Colors;
+    fontSize?: number;
+    borderRadius?: number;
```
Comment (BrandonDenham) [inline src/components/badge/badge.types.ts:8]: delete the nonused props from here please

File: src/components/badge/badge.types.ts [hunk 14]
```diff
@@ -0,0 +1,10 @@
+import { Colors } from '../../shared/constants/colors';
+import { SyntheticEvent } from 'react';
+export default interface BadgeProps {
+    label?: string;
+    backgroundColor?: string;
+    color?: Colors;
+    fontSize?: number;
```
Comment (BrandonDenham) [inline src/components/badge/badge.types.ts:7]: delete

File: src/components/badge/badge.types.ts [hunk 15]
```diff
@@ -0,0 +1,10 @@
+import { Colors } from '../../shared/constants/colors';
+import { SyntheticEvent } from 'react';
+export default interface BadgeProps {
+    label?: string;
```
Comment (BrandonDenham) [inline src/components/badge/badge.types.ts:4]: this should be required i think

**Review by BrandonDenham — APPROVED** [review]
some comments, please address then this is good to merge

File: src/components/badge/badge.stories.tsx (hunk 11 above)
Comment (parrajulian) [reply src/components/badge/badge.stories.tsx]: done

File: src/components/badge/badge.tsx (hunk 12 above)
Comment (parrajulian) [reply src/components/badge/badge.tsx]: on sync with the spec

Comment (parrajulian) [reply src/components/badge/badge.tsx]: modified

File: src/components/badge/badge.types.ts (hunk 13 above)
Comment (parrajulian) [reply src/components/badge/badge.types.ts]: deleted

File: src/components/badge/badge.types.ts (hunk 14 above)
Comment (parrajulian) [reply src/components/badge/badge.types.ts]: done

File: src/components/badge/badge.types.ts (hunk 15 above)
Comment (parrajulian) [reply src/components/badge/badge.types.ts]: label changed to text and is now required

**Pull request merged**